unstructured_ingest/v2/processes/connectors/fsspec/fsspec.py

```diff
@@ -207,12 +210,34 @@ def fs(self) -> "AbstractFileSystem":
             **self.connection_config.get_access_config(),
         )
 
+    def handle_directory_download(self, lpath: Path) -> None:
+        # When the extension has odd characters in it (i.e. ???,gz), it
+        # gets downloaded in a new directory rather than as a file. This reconciles
+        # that with what is expected.
```

Comment:
Can we modify this comment? Downloading into a directory also happens when the '?' appears elsewhere in the name, not just in the extension.

```suggestion
        # If the object's name contains certain characters (i.e. '?'), it
        # gets downloaded into a new directory of the same name. This
        # reconciles that with what is expected, which is to download it 
        # as a file that is not within a directory.